packages/core/src/dispatcher.ts

```diff
@@ -123,9 +126,13 @@ export default class Dispatcher extends EventEmitter {
       const cid = await this._ipfs.dag.put(jws, { format: 'dag-jose', hashAlg: 'sha2-256' })
       // put the payload into the ipfs dag
       await this._ipfs.block.put(linkedBlock, { cid: jws.link.toString() })
+      await this._restrictRecordSize(jws.link.toString())
```

Comment:
Could we enforce the size limit before storing the record in IPFS?  Otherwise we spend a lot of time uploading a large file over the network for nothing.  This would however require calculating the object size in memory rather than relying on IPFS's `stat` method...

Reply:
I guess we can use IPLD for calculating the size of the record but I have to check that.

Reply:
Let's investigate this as part of [621](https://github.com/ceramicnetwork/js-ceramic/issues/621) story.